src/core/border_router/routing_manager.cpp

```diff
@@ -260,20 +268,41 @@ void RoutingManager::RecvIcmp6Message(uint32_t            aInfraIfIndex,
     }
 }
 
+void RoutingManager::HandleInfraIfStateChanged(uint32_t            aInfraIfIndex,
+                                               bool                aIsRunning,
+                                               const Ip6::Address *aLinkLocalAddress)
+{
+    VerifyOrExit(IsInitialized() && aInfraIfIndex == mInfraIfIndex);
+    VerifyOrExit(aLinkLocalAddress == nullptr || aLinkLocalAddress->IsLinkLocal());
+
+    otLogInfoBr("infra interface state changed: %s, link-local-addr=%s", aIsRunning ? "RUNNING" : "NOT RUNNING",
+                (aLinkLocalAddress != nullptr) ? aLinkLocalAddress->ToString().AsCString() : "(null)");
+
+    mInfraIfIsRunning = aIsRunning;
+    if (aLinkLocalAddress == nullptr)
+    {
+        mInfraIfLinkLocalAddress.Clear();
+    }
+    else
+    {
+        mInfraIfLinkLocalAddress = *aLinkLocalAddress;
+    }
+
+    EvaluateState();
+
+exit:
+    return;
+}
+
 void RoutingManager::HandleNotifierEvents(Events aEvents)
 {
     VerifyOrExit(IsInitialized() && IsEnabled());
 
     if (aEvents.Contains(kEventThreadRoleChanged))
     {
-        if (Get<Mle::MleRouter>().IsAttached())
-        {
-            Start();
-        }
-        else
-        {
-            Stop();
-        }
+        VerifyOrExit(Get<Mle::MleRouter>().IsAttached() != mIsAttached);
+        mIsAttached = Get<Mle::MleRouter>().IsAttached();
```

Comment:
```suggestion
```
why do we need to save `Get<Mle::MleRouter>().IsAttached()` to `mIsAttached` if we can just call `Get<Mle::MleRouter>().IsAttached()` whenever we want?

Reply:
Not really necessary, just to be symmetrical with the `mInfraIfisRunning` state. I am ok to remove it.